backend/specs/akvo/lumen/specs/visualisation.clj

```diff
@@ -68,63 +93,163 @@
+(s/def ::sizeLabel (s/nilable string?))
+(s/def ::categoryLabelFromUser (s/nilable boolean?))
+(s/def ::sizeLabelFromUser boolean?)
+(s/def ::scatter.s/metricColumnY (s/nilable ::db.dsv.column.s/columnName))
+(s/def ::scatter.s/metricColumnX (s/nilable ::db.dsv.column.s/columnName))
+(s/def ::scatter.s/metricAggregation (s/nilable ::aggregation.s/metricAggregation))
+(s/def ::scatter.s/spec (s/merge ::base-spec
+                                 (s/keys :req-un [::datapointLabelColumn
+                                                  ::scatter.s/metricColumnY ::scatter.s/metricColumnX
+                                                  ::bucketColumn
+                                                  ::scatter.s/metricAggregation
+                                                  ::axisLabelXFromUser ::axisLabelX
+                                                  ::axisLabelYFromUser ::axisLabelY
+                                                  ]
+                                         :opt-un [::metricColumnSize
+                                                  ::bucketColumnCategory
+                                                  ::sizeLabelFromUser
+                                                  ::categoryLabelFromUser
+                                                  ::categoryLabel
+                                                  ::sizeLabel])))
 
-(defmethod vis "polararea"  [_]
-  (s/merge ::base-viz #_(s/keys :req-un [::area.s/spec])))
+(defmethod vis "scatter"  [_]
+  (s/merge ::base-viz (s/keys :req-un [::scatter.s/spec])))
+
+(create-ns  'akvo.lumen.specs.visualisation.bubble)
+(alias 'bubble.s 'akvo.lumen.specs.visualisation.bubble)
+
+(s/def ::metricLabelFromUser boolean?)
+(s/def ::bucketLabel (s/nilable string?))
+(s/def ::metricLabel (s/nilable string?))
+(s/def ::metricColumn (s/nilable ::db.dsv.column.s/columnName))
+(s/def ::truncateSize (s/nilable string?))
+
+(s/def ::bubble.s/spec (s/merge ::base-spec
+                                (s/keys :req-un [::metricLabelFromUser
+                                                 ::aggregation.s/metricAggregation
+                                                 ::metricLabel
+                                                 ::metricColumn
+                                                 ::truncateSize]
+                                        :opt-un [::bucketColumn
+                                                 ::bucketLabel])))
 
-(defmethod vis "donut"  [_]
-  (s/merge ::base-viz #_(s/keys :req-un [::area.s/spec])))
+(defmethod vis "bubble"  [_]
+  (s/merge ::base-viz (s/keys :req-un [::bubble.s/spec])))
+
+(create-ns  'akvo.lumen.specs.visualisation.pivot)
+(alias 'pivot.s 'akvo.lumen.specs.visualisation.pivot)
+
+(s/def ::hideColumnTotals boolean?)
+(s/def ::hideRowTotals boolean?)
+(s/def ::rowColumn (s/nilable ::db.dsv.column.s/columnName))
+
+(defn str-nat-int? [v]
+  (when (some? v)
+    (try
+      (nat-int? (read-string v))
```

Comment:
Let's use a different strategy for converting to number, `read-string` should be avoided IMHO.

Reply:
changed https://github.com/akvo/akvo-lumen/pull/2647/commits/b8522d02ff6f7a3356d715413c54436f67886ee5

Reply:
and https://github.com/akvo/akvo-lumen/pull/2647/commits/84ea1be0d56de8c507501b43f149aab8bbe27edd